index.js

```diff
@@ -1,43 +1,52 @@
 const http = require('http');
 const events = require('events');
-const {EventEmitter} = events;
+const { EventEmitter } = events;
 const express = require('express');
 const ws = require('ws');
 const engine = require('./build/Release/qr.node');
 
+async function initQr(qrEmitter, locomotionEmitter) {
+  return new Promise((resolve, reject) => {
+    const resultCode = engine.initVr();
+
+    // Reject if there was an error initializing OpenVR
+    // See https://github.com/ValveSoftware/openvr/wiki/HmdError for error codes
+    if (resultCode !== 0) {
+      console.error(resultCode === 100 ? 'OpenVR Installation not found' : 'Error initialising OpenVR', resultCode);
+      reject(resultCode);
+      return;
+    }
+
+    engine.createQrEngine(qrCode => qrEmitter.emit('qrCode', qrCode));
+
+    engine.createLocomotionEngine(locomotionInput =>
+      locomotionEmitter.emit('locomotionInput', locomotionInput)
+    );
+    engine.startThread();
+    resolve();
+  });
+}
+
 async function start({
   port = 8000,
   key,
 } = {}) {
-  let initialized = false;
   let live = false;
-  let qrEmitter = null;
-  let locomotionEmitter = null;
+  let qrEmitter = new EventEmitter();
+  let locomotionEmitter = new EventEmitter();
+
   const presenceWss = new ws.Server({
     noServer: true,
   });
-  presenceWss.on('connection', (s, req) => {
-    if (!initialized) {
-      engine.initVr();
-      qrEmitter = new EventEmitter();
-      engine.createQrEngine(qrCode => {
-        qrEmitter.emit('qrCode', qrCode);
-      });
-      locomotionEmitter = new EventEmitter();
-      engine.createLocomotionEngine(locomotionInput => {
-        locomotionEmitter.emit('locomotionInput', locomotionInput);
-      });
-      engine.startThread();
 
-      initialized = true;
-    }
+  presenceWss.on('connection', (s, req) => {
     if (!live) {
       live = true;
 
       let authed = typeof key !== 'string';
       s.on('message', s => {
         const j = JSON.parse(s);
-        const {method} = j;
+        const { method } = j;
```

Comment:
Is this being done by lint rules?

Reply:
It was! I've added the actual eslint semistandard config now and re-linted the code

Reply:
Is there a way to not add these spaces in the linter for this PR?

There might be a time to re-format files, but I don't think it should be done with spacing changes sprinkled into unrelated PR's.

Reply:
Sure, I've removed the spaces